packages/arb-token-bridge-ui/src/token-bridge-sdk/EthDepositStarter.ts

```diff
@@ -0,0 +1,114 @@
+import { EthBridger } from '@arbitrum/sdk'
+import { BigNumber } from 'ethers'
+import {
+  ApproveNativeCurrencyProps,
+  BridgeTransferStarter,
+  BridgeTransferStarterProps,
+  RequiresNativeCurrencyApprovalProps,
+  TransferEstimateGas,
+  TransferProps,
+  TransferType
+} from './BridgeTransferStarter'
+import { getAddressFromSigner, percentIncrease } from './utils'
+import { depositEthEstimateGas } from '../util/EthDepositUtils'
+import { fetchErc20Allowance } from '../util/TokenUtils'
+
+export class EthDepositStarter extends BridgeTransferStarter {
+  public transferType: TransferType = 'eth_deposit'
+
+  constructor(props: BridgeTransferStarterProps) {
+    super(props)
+  }
+
+  public async requiresNativeCurrencyApproval({
+    amount,
+    signer
+  }: RequiresNativeCurrencyApprovalProps) {
+    const address = await getAddressFromSigner(signer)
+
+    const ethBridger = await EthBridger.fromProvider(
+      this.destinationChainProvider
+    )
+    const { l2Network } = ethBridger
+
+    if (typeof l2Network.nativeToken === 'undefined') {
+      return false // no native currency found for the network
+    }
+
+    const customFeeTokenAllowanceForInbox = await fetchErc20Allowance({
+      address: l2Network.nativeToken,
+      provider: this.sourceChainProvider,
+      owner: address,
+      spender: l2Network.ethBridge.inbox
+    })
+
+    // We want to bridge a certain amount of the custom fee token, so we have to check if the allowance is enough.
+    return customFeeTokenAllowanceForInbox.lt(amount)
+  }
+
+  public async approveNativeCurrency({ signer }: ApproveNativeCurrencyProps) {
+    const ethBridger = await EthBridger.fromProvider(
+      this.destinationChainProvider
+    )
+    const approveCustomFeeTokenTx = await ethBridger.approveFeeToken({
+      l1Signer: signer
+    })
+    await approveCustomFeeTokenTx.wait()
```

Comment:
(applies to all) we should return the transaction object instead of the transaction receipt. this leaves more flexibility to the consumer. right now the method wouldn't exit until the transaction got included in a block which can take a while

```suggestion
    return ethBridger.approveFeeToken({
      l1Signer: signer
    })
```

Reply:
Good point. Updated.